control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3217,13 +3220,29 @@ func (r *HostedControlPlaneReconciler) removeCloudResources(ctx context.Context,
 	log.Info("Removing cloud resources")
 
 	// check if resources have been destroyed
-	if meta.IsStatusConditionTrue(hcp.Status.Conditions, string(hyperv1.CloudResourcesDestroyed)) {
+	resourcesDestroyedCond := meta.FindStatusCondition(hcp.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+	if resourcesDestroyedCond != nil && resourcesDestroyedCond.Status == metav1.ConditionTrue {
```

Comment:
isn't `resourcesDestroyedCond != nil && resourcesDestroyedCond.Status == metav1.ConditionTrue`
The exact underlying impl of `IsStatusConditionTrue` (which calls `IsStatusConditionPresentAndEqual`)? why the change?

Reply:
The only reason is that it's used below (line 3236), but we can re-fetch if that looks better.